Engine/script/cc_instance.cpp

```diff
@@ -125,33 +125,33 @@ const ScriptCommandInfo sccmd_info[CC_NUM_SCCMDS] =
-    ScriptCommandInfo( SCMD_FADDREG         , "f.add"             , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_FSUBREG         , "f.sub"             , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_FGREATER        , "f.gt"              , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_FLESSTHAN       , "f.lt"              , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_FGTE            , "f.gte"             , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_FLTE            , "f.lte"             , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_CHECKBOUNDS     , "assertlte"         , 2, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMWRITEPTR     , "ptrset"            , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMREADPTR      , "ptrget"            , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMZEROPTR      , "ptrzero"           , 0, kScOpNoArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMINITPTR      , "ptrinit"           , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_LOADSPOFFS      , "ptrstrack"         , 1, kScOpNoArgIsReg ),
+    ScriptCommandInfo( SCMD_CHECKNULL       , "ptrassert"         , 0, kScOpNoArgIsReg ),
+    ScriptCommandInfo( SCMD_FADD            , "faddi"             , 2, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_FSUB            , "fsubi"             , 2, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_FMULREG         , "fmul"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FDIVREG         , "fdiv"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FADDREG         , "fadd"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FSUBREG         , "fsub"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FGREATER        , "fgt"               , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FLESSTHAN       , "flt"               , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FGTE            , "fgte"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_FLTE            , "flte"              , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_ZEROMEMORY      , "zeromem"           , 1, kScOpNoArgIsReg ),
-    ScriptCommandInfo( SCMD_CREATESTRING    , "newstring"         , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_STRINGSEQUAL    , "strcmp"            , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_STRINGSNOTEQ    , "strnotcmp"         , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_CHECKNULLREG    , "checknull"         , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_CREATESTRING    , "newstr"            , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_STRINGSEQUAL    , "streq"             , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_STRINGSNOTEQ    , "strne"             , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_CHECKNULLREG    , "assert"            , 1, kScOpOneArgIsReg ),
```

Comment:
This instruction specifically checks for null registers, but `assert` sounds too general. At least `checknull` told us what it did.

Reply:
assert precisely documents what the instruction does. if the value is 0, is raises an error, just like the assert everybody knows.
in C you'd write `assert(REG);` and `assert reg` in asm comes closest.

Reply:
Not quite, it checks if the pointer value is null. I'm not sure it's meant to be used for registers with non-pointer values.

Reply:
Technically it checks if the any value is 0. But engine is currently hardcoded to report a "string is null error", maybe because this was only case it was used so far.

Reply:
I guess there's a conflict between what the operation is actually doing and what it was used for. E.g. compare with SCMD_CHECKBOUNDS, which is practically testing if integer is between 0 and X, but was used specifically to test array index.

Reply:
> things may become complicated if these operations will be used for other purposes though

yes, possibly. though i don't see anybody writing asm directly (except for agsutils users, or maybe a future script compiler), so this shouldn't be an issue.